packages/components/search/src/search.js

```diff
@@ -0,0 +1,108 @@
+import { TSElement, unsafeCSS, html, customElementDefineHelper, helpers } from '@tradeshift/elements';
+import css from './search.css';
+
+import '@tradeshift/elements.icon';
+
+customElementDefineHelper(
+	'ts-search',
+	class extends TSElement {
+		static get styles() {
+			return [TSElement.styles, unsafeCSS(css)];
+		}
+
+		static get properties() {
+			return {
+				autofocus: { type: Boolean, reflect: true },
+				dir: { type: String, reflect: true },
+				focused: { type: Boolean, reflect: true },
+				idleTime: { type: Number, attribute: 'idle-time', reflect: true },
+				placeholder: { type: String, reflect: true },
+				value: { type: String, reflect: true }
+			};
+		}
+
+		constructor() {
+			super();
+			this.idleTime = 300;
+			this.placeholder = 'Search...';
+			this.value = '';
+			this.dispatchIdleEvent = helpers.debounceEvent(() => this.dispatchCustomEvent('idle', this.value), this.idleTime);
+		}
+
+		get direction() {
+			return this.dir ? this.dir : this.bodyDir;
+		}
+
+		get showCancel() {
+			return Boolean(this.value);
+		}
+
+		attributeChangedCallback(name, oldVal, newVal) {
+			super.attributeChangedCallback(name, oldVal, newVal);
+			switch (name) {
+				case 'idle-time':
+					this.dispatchIdleEvent = helpers.debounceEvent(() => this.dispatchCustomEvent('idle', this.value), newVal);
+					break;
+				case 'value':
+					this.dispatchCustomEvent('change', newVal);
+					this.dispatchIdleEvent();
+					break;
+				case 'focused':
+					if (this.focused) {
+						this.shadowRoot.getElementById('input').focus();
+					}
+					break;
+				default:
+				// do nothing
+			}
+		}
+
+		handleInput(e) {
+			this.value = e.target.value;
+		}
+
+		handleKeyDown(e) {
+			if (e.key === 'Enter') {
```

Comment:
Not sure if it's causing an issue, @zdlm where we can find the list of the supported browsers by Tradeshift?
https://caniuse.com/#feat=keyboardevent-key